### PR TITLE
Fix Wijkander's friday bug

### LINKED ISSUE #83: Wijkander's has a Friday bug
Mat shows the menu of the whole week on fridays, this is not intended behaviour. 

Attached to the issue is a screenshot and a gzipped version of today's Wijkander's page.

wijkanders-bug

[Lunch menu - Wijkanders.html.gz](https://github.com/dtekcth/mat-chalmers/files/3190955/Lunch.menu.-.Wijkanders.html.gz)

### LINKED ISSUE #88: Wijkanders breaks on fridays?
Wijkanders at least broke today, and today it is friday. The menu should only show today's menu, not the menu for the whole week. See attached gzipped HTML for a testcase.

wijkanderstrasig

[Lunchmeny - wijkanders.se.html.gz](https://github.com/dtekcth/mat-chalmers/files/3609065/Lunchmeny.-.wijkanders.se.html.gz)

### PR DESCRIPTION
**Pull request by Rembane**
Before this fix the parser would take all meals from the next week too
but now we stop at the next date instead of the next consecutive date.
Hopefully this makes the parser mure resilient to for instance easter
and fridays.

There are two tests in this commit, to verify that the parser can
handle the input data that broke the parser before.

This commit closes #83 and it also closes #88.

### CODE REVIEW

**Comment by Rembane** [comment]
There's some seriously confusing code in this commit, so feel free to post suggestions on how to make it clearer.

File: src/Model/Wijkanders.hs [hunk 1]
```diff
@@ -48,13 +50,16 @@ import           Util                                     ( menusToEitherNoLunch
...
-        )
+-- | Looks for strings looking like dates, dd/mm where d and m are digits.
+-- ..and gives them in another order to play nice with the
+-- YearMonthDay constructor.
+hasDate :: ByteString -> Maybe (Months, Days)
+hasDate = maybeResult . parse
+  (   ((.) . (.)) swap (,)
+  <$> (skipMany (skip (not . W8.isDigit)) *> integerParser)
+  <*> (string (B8.pack "/") *> integerParser)
+  )
```
Comment (adamse) [inline src/Model/Wijkanders.hs]: if this is some kind of computation that is a monad: use do notation and get rid of all the point free stuff :)

Comment (Rembane) [reply src/Model/Wijkanders.hs]: It isn't a computation imo, but on the other hand, it is quote convoluted and it's more about my fascination for pointless style than actually creating readable code. :) So I'll probably rewrite it in a more readable fashion in a few days.

Comment (Rembane) [reply src/Model/Wijkanders.hs]: I find writing parsers (using parser combinators) that are readable quite hard.

Comment (Rembane) [reply src/Model/Wijkanders.hs]: So here's another version of the same code: https://github.com/dtekcth/mat-chalmers/pull/89/commits/dc42e1d4a69fb090fbdd44bbf27a11ec0498090f#diff-9167874cd996e771711a5e4343d80061R56

Comment (Rembane) [reply src/Model/Wijkanders.hs]: And yet another one. This was interesting. https://github.com/dtekcth/mat-chalmers/pull/89/commits/8c264cc38f5034c6b20d68ed5ddaf6088080f5b7#diff-9167874cd996e771711a5e4343d80061R56

**Review by Jassob — APPROVED** [review]
I like the version from "More words, is this better?", let's keep that.

Please fixup/squash the commits as appropriate and then we're good to go!

**Comment by Rembane** [comment]
> I like the version from "More words, is this better?", let's keep that.
> 
> Please fixup/squash the commits as appropriate and then we're good to go!

It is done. I'm now waiting for the CIs to do their thing... then rebasing will happen!

**Comment by adamse** [comment]
I wasn't clear, but this is the solution, imho:

```
do 
   _ <- skipMany (skip (not . W8.isDigit))
  day <- integerParser
  _  <- string (B8.pack "/")
  month <- integerParser
  pure (month, day)
```

**Comment by Jassob** [comment]
I didn't think the do-version was that much clearer, which why I approved
the applicative-version instead.

However, given that there might be a day where we want to hand over the
code to another set of Haskell-friendly students, we probably should start
writing more do-notation as it is easier to get started with.

// Jsb

Den mån 23 sep. 2019 kl 23:27 skrev Adam Sandberg Eriksson <
notifications@github.com>:

**Pull request merged**